README.md

```diff
@@ -32,90 +32,131 @@ extern crate rspec;
 
 You can see complete examples in the [`examples/`](https://github.com/mackwic/rspec/tree/master/examples) directory.
 
-You can now use rspec in your unit tests, this example will use the `cargo test`
-runner:
 
 ```rust
-fn add(x: u32, y: u32) -> u64 {
-    x + y
-}
+#[macro_use(rspec_run)]
+extern crate rspec;
 
-#[test]
-fn test_add() {
-    rdescribe("add", |ctx| {
-        ctx.describe("0 <= x + y <= u32::MAX", |ctx| {
-            ctx.it("2 + 4 = 6", || {
-                assert_eq!(6, add(2, 4))
+pub fn main() {
+    // Use a local struct to provide the test contexts with an environment.
+    // The environment will contain the subject that is to be tested
+    // along with any additional data you might need during the test run:
+    #[derive(Clone, Default, Debug)]
+    struct Environment {
+        // ...
+    }
```

Comment:
I think the Environment is an advanced feature that should not be used by default for every test. Maybe split the example in 2, one really simple, and another that showcase the Environment use ?